models/Semester.js

```diff
@@ -0,0 +1,90 @@
+const HttpError = require('http-errors');
+const log = require('loglevel');
+const {
+  db
+} = require('../services/database');
+const {
+  whereParams,
+  updateValues
+} = require('../services/sqltools');
+
+
+//  TEMP MADE FOR TESTING EDIT
+// if found return { ... }
+// if not found return {}
+// if db error, db.query will throw a rejected promise
+async function findOne(criteria) {
+
+  const {
+    text,
+    params
+  } = whereParams(criteria);
+
+  const res = await db.query(`SELECT * from "semester" ${text} LIMIT 1;`, params);
+
+  if (res.rows.length > 0) {
+    log.debug(`Successfully found semester from db with criteria: ${text}, ${JSON.stringify(params)}`);
+    return res.rows[0];
+  }
+  log.debug(`No semester found in db with criteria: ${text}, ${JSON.stringify(params)}`);
+  return {};
+}
+
+
+// Edit given course's attributes
+// if successful update record in database, return row modified 'res'
+// if successful, but no row updates/returned, throw error
+// if not enough parameters, throw error
+// otherwise throw error
+async function editSemester(id, resultSemester) {
+  if (id && resultSemester) {
+
+    if (!(resultSemester.year == null) || !(resultSemester.type == null)) {
+
+      const newSemesterJSON = {}
+
+      if (!(resultSemester.year == null))
+        newSemesterJSON.year = resultSemester.year
+
+      if (!(resultSemester.type == null))
```

Comment:
Try to use curly braces on your if statements for readability if you could.

Reply:
Obviously if it is one line that is fine but multiple lines in your if statements should be surrounded by braces for best practice.